services/galley/test/integration/API/CustomBackend.hs

```diff
@@ -87,3 +88,25 @@ getByDomainDeleted = do
   get (galley . path "/custom-backend/by-domain/domain.no3") !!! do
     const 404 === statusCode
     const ("domain.no3" :: ByteString) =~= (cs . fold . responseBody)
+
+getByDomainIsCaseInsensitive :: TestM ()
+getByDomainIsCaseInsensitive = do
+  galley <- view tsGalley
+  let jsonBody :: Value
+      jsonBody =
+        [aesonQQ|{
+          "config_json_url": "https://wire-rest.https.no2.com/config.json",
+          "webapp_welcome_url": "https://app.wire.no2.com/"
+          }|]
+  put (galley . path "/i/custom-backend/by-domain/Ab.Ab" . json jsonBody)
+    !!! const 201 === statusCode
+  get (galley . path "/custom-backend/by-domain/aB.aB") !!! do
+    const 200 === statusCode
+    const jsonBody === responseJsonUnsafe
+  delete (galley . path "/i/custom-backend/by-domain/AB.ab")
+    !!! const 200 === statusCode
+  -- not just the matching is done in a case-insensitive way,
+  -- but we actually store the normalized domain.
+  get (galley . path "/custom-backend/by-domain/Ab.Ab") !!! do
+    const 404 === statusCode
```

Comment:
```suggestion
    const 200 === statusCode
```

As I understand it, this is exactly the problem.  The client does a lookup by domain, and shouldn't have to normalize the lookup, but interpret the string case-insensitively.

I haven't checked the context, though.

Reply:
This is a 404 because we deleted the entry just before. There is similar call with 200 a few lines above.